"Distribution" function doesn't work with BigQuery BIGINTEGER/BIGDECIMAL

**Describe the bug**
In my current setup on 0.45.3 the distribution quick function is broken and return the following error: 


> Unable to bin Field without a min/max value

![Screenshot 2023-02-23 at 14 18 21](https://user-images.githubusercontent.com/113892067/220917696-83e38fc7-34f5-4229-b069-aff9fa7537de.png)
![Screenshot 2023-02-23 at 14 18 52](https://user-images.githubusercontent.com/113892067/220917832-f5be881a-ff9f-4c35-a042-968e83f787d4.png)
![Screenshot 2023-02-23 at 14 19 21](https://user-images.githubusercontent.com/113892067/220918008-ca891405-41ff-4a4d-b805-bde3551983fe.png)

**Logs**
```
react-dom.production.min.js:49 Uncaught Error: Minified React error #188; visit https://reactjs.org/docs/error-decoder.html?invariant=188 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at react-dom.production.min.js:49:67
    at it (react-dom.production.min.js:50:217)
    at t.findDOMNode (react-dom.production.min.js:289:336)
    at h.value (ExplicitSize.jsx:39:25)
    at ExplicitSize.jsx:126:30
    at l (throttle.js:32:21)
(anonymous) @ react-dom.production.min.js:49
it @ react-dom.production.min.js:50
t.findDOMNode @ react-dom.production.min.js:289
value @ ExplicitSize.jsx:39
(anonymous) @ ExplicitSize.jsx:126
l @ throttle.js:32
setTimeout (async)
value @ ExplicitSize.jsx:51
us @ react-dom.production.min.js:212
yc @ react-dom.production.min.js:255
t.unstable_runWithPriority @ scheduler.production.min.js:19
Yi @ react-dom.production.min.js:122
Oc @ react-dom.production.min.js:248
cc @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Yi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
D @ react-dom.production.min.js:287
Zt @ react-dom.production.min.js:68
settings.js:110 Error getting setting graph.x_axis._is_histogram TypeError: Cannot read properties of undefined (reading 'binning_info')
    at Object.getDefault (graph.js:495:15)
    at et (settings.js:101:39)
    at Xe (settings.js:42:5)
    at v (visualization.js:76:10)
    at LineAreaBarChart.jsx:425:20
    at LineAreaBarChart.jsx:173:9
    at Array.map (<anonymous>)
    at Function.value (LineAreaBarChart.jsx:172:17)
    at P (index.js:74:30)
    at selectors.js:675:18
et @ settings.js:110
Xe @ settings.js:42
v @ visualization.js:76
(anonymous) @ LineAreaBarChart.jsx:425
(anonymous) @ LineAreaBarChart.jsx:173
value @ LineAreaBarChart.jsx:172
P @ index.js:74
(anonymous) @ selectors.js:675
(anonymous) @ index.js:76
(anonymous) @ index.js:36
(anonymous) @ index.js:90
(anonymous) @ index.js:36
(anonymous) @ index.js:86
(anonymous) @ index.js:36
(anonymous) @ index.js:86
(anonymous) @ index.js:36
(anonymous) @ index.js:86
(anonymous) @ index.js:36
(anonymous) @ index.js:86
(anonymous) @ index.js:36
(anonymous) @ index.js:86
(anonymous) @ index.js:36
(anonymous) @ QueryBuilder.jsx:131
r @ wrapMapToProps.js:41
f @ selectorFactory.js:44
(anonymous) @ selectorFactory.js:63
run @ connectAdvanced.js:22
c.componentWillReceiveProps @ connectAdvanced.js:154
_o @ react-dom.production.min.js:135
Fa @ react-dom.production.min.js:179
As @ react-dom.production.min.js:263
Mc @ react-dom.production.min.js:246
gc @ react-dom.production.min.js:246
cc @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Yi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
rc @ react-dom.production.min.js:230
enqueueSetState @ react-dom.production.min.js:132
z.setState @ react.production.min.js:12
c.onStateChange @ connectAdvanced.js:207
notify @ Subscription.js:23
t.notifyNestedSubs @ Subscription.js:62
c.onStateChange @ connectAdvanced.js:204
h @ createStore.js:172
(anonymous) @ middleware.js:22
(anonymous) @ index.js:28
(anonymous) @ store.js:22
dispatch @ applyMiddleware.js:35
action @ store.js:17
(anonymous) @ querying.js:213
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
oe @ object-detail.js:158
a @ object-detail.js:158
(anonymous) @ object-detail.js:158
(anonymous) @ object-detail.js:158
(anonymous) @ querying.js:178
(anonymous) @ store.js:20
dispatch @ applyMiddleware.js:35
(anonymous) @ querying.js:144
Promise.then (async)
(anonymous) @ querying.js:135
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
oe @ object-detail.js:158
a @ object-detail.js:158
(anonymous) @ object-detail.js:158
(anonymous) @ object-detail.js:158
(anonymous) @ querying.js:99
(anonymous) @ store.js:20
dispatch @ applyMiddleware.js:35
(anonymous) @ initializeQB.ts:351
setTimeout (async)
(anonymous) @ initializeQB.ts:350
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
lr @ parameterUtils.ts:112
a @ parameterUtils.ts:112
Promise.then (async)
lr @ parameterUtils.ts:112
a @ parameterUtils.ts:112
Promise.then (async)
lr @ parameterUtils.ts:112
a @ parameterUtils.ts:112
(anonymous) @ parameterUtils.ts:112
(anonymous) @ parameterUtils.ts:112
Sr @ initializeQB.ts:239
(anonymous) @ initializeQB.ts:369
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
lr @ parameterUtils.ts:112
a @ parameterUtils.ts:112
(anonymous) @ parameterUtils.ts:112
(anonymous) @ parameterUtils.ts:112
(anonymous) @ initializeQB.ts:366
(anonymous) @ store.js:20
(anonymous) @ bindActionCreators.js:3
(anonymous) @ QueryBuilder.jsx:309
(anonymous) @ use-on-mount.ts:5
ls @ react-dom.production.min.js:211
wc @ react-dom.production.min.js:257
t.unstable_runWithPriority @ scheduler.production.min.js:19
Yi @ react-dom.production.min.js:122
_c @ react-dom.production.min.js:257
cc @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Yi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
D @ react-dom.production.min.js:287
Zt @ react-dom.production.min.js:68
settings.js:110 Error getting setting graph.x_axis._is_histogram TypeError: Cannot read properties of undefined (reading 'binning_info')
    at Object.getDefault (graph.js:495:15)
    at et (settings.js:101:39)
    at Xe (settings.js:42:5)
    at v (visualization.js:76:10)
    at selectors.js:697:23
    at index.js:76:25
    at index.js:36:25
    at index.js:90:33
    at index.js:36:25
    at index.js:86:37
et @ settings.js:110
Xe @ settings.js:42
v @ visualization.js:76
(anonymous) @ selectors.js:697
(anonymous) @ index.js:76
(anonymous) @ index.js:36
(anonymous) @ index.js:90
(anonymous) @ index.js:36
(anonymous) @ index.js:86
(anonymous) @ index.js:36
(anonymous) @ index.js:86
(anonymous) @ index.js:36
(anonymous) @ index.js:86
(anonymous) @ index.js:36
(anonymous) @ QueryBuilder.jsx:131
r @ wrapMapToProps.js:41
f @ selectorFactory.js:44
(anonymous) @ selectorFactory.js:63
run @ connectAdvanced.js:22
c.componentWillReceiveProps @ connectAdvanced.js:154
_o @ react-dom.production.min.js:135
Fa @ react-dom.production.min.js:179
As @ react-dom.production.min.js:263
Mc @ react-dom.production.min.js:246
gc @ react-dom.production.min.js:246
cc @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Yi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
rc @ react-dom.production.min.js:230
enqueueSetState @ react-dom.production.min.js:132
z.setState @ react.production.min.js:12
c.onStateChange @ connectAdvanced.js:207
notify @ Subscription.js:23
t.notifyNestedSubs @ Subscription.js:62
c.onStateChange @ connectAdvanced.js:204
h @ createStore.js:172
(anonymous) @ middleware.js:22
(anonymous) @ index.js:28
(anonymous) @ store.js:22
dispatch @ applyMiddleware.js:35
action @ store.js:17
(anonymous) @ querying.js:213
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
oe @ object-detail.js:158
a @ object-detail.js:158
(anonymous) @ object-detail.js:158
(anonymous) @ object-detail.js:158
(anonymous) @ querying.js:178
(anonymous) @ store.js:20
dispatch @ applyMiddleware.js:35
(anonymous) @ querying.js:144
Promise.then (async)
(anonymous) @ querying.js:135
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
oe @ object-detail.js:158
a @ object-detail.js:158
(anonymous) @ object-detail.js:158
(anonymous) @ object-detail.js:158
(anonymous) @ querying.js:99
(anonymous) @ store.js:20
dispatch @ applyMiddleware.js:35
(anonymous) @ initializeQB.ts:351
setTimeout (async)
(anonymous) @ initializeQB.ts:350
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
lr @ parameterUtils.ts:112
a @ parameterUtils.ts:112
Promise.then (async)
lr @ parameterUtils.ts:112
a @ parameterUtils.ts:112
Promise.then (async)
lr @ parameterUtils.ts:112
a @ parameterUtils.ts:112
(anonymous) @ parameterUtils.ts:112
(anonymous) @ parameterUtils.ts:112
Sr @ initializeQB.ts:239
(anonymous) @ initializeQB.ts:369
u @ runtime.js:63
(anonymous) @ runtime.js:293
(anonymous) @ runtime.js:118
lr @ parameterUtils.ts:112
a @ parameterUtils.ts:112
(anonymous) @ parameterUtils.ts:112
(anonymous) @ parameterUtils.ts:112
(anonymous) @ initializeQB.ts:366
(anonymous) @ store.js:20
(anonymous) @ bindActionCreators.js:3
(anonymous) @ QueryBuilder.jsx:309
(anonymous) @ use-on-mount.ts:5
ls @ react-dom.production.min.js:211
wc @ react-dom.production.min.js:257
t.unstable_runWithPriority @ scheduler.production.min.js:19
Yi @ react-dom.production.min.js:122
_c @ react-dom.production.min.js:257
cc @ react-dom.production.min.js:239
(anonymous) @ react-dom.production.min.js:123
t.unstable_runWithPriority @ scheduler.production.min.js:19
Yi @ react-dom.production.min.js:122
Zi @ react-dom.production.min.js:123
Ki @ react-dom.production.min.js:122
D @ react-dom.production.min.js:287
Zt @ react-dom.production.min.js:68
VisualizationError.jsx:96 error Unable to bin Field without a min/max value
VisualizationError.jsx:96 error Unable to bin Field without a min/max value
VisualizationError.jsx:96 error Unable to bin Field without a min/max value
VisualizationError.jsx:96 error Unable to bin Field without a min/max value
VisualizationError.jsx:96 error Unable to bin Field without a min/max value
```

```

> [a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 DEBUG metabase.server.middleware.log POST /api/dataset 202 [ASYNC: completed] 1.0 s (61 DB calls) App DB connections: 1/15 Jetty threads: 2/50 (5 idle, 0 queued) (98 total active threads) Queries in flight: 1 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:05+01:00 DEBUG metabase.server.middleware.log POST /api/dataset 202 [ASYNC: completed] 575.1 ms (23 DB calls) App DB connections: 1/15 Jetty threads: 2/50 (5 idle, 0 queued) (98 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:23+01:00 ERROR metabase.query-processor.middleware.catch-exceptions Error processing query: Unable to bin Field without a min/max value
{:database_id 3,
 :started_at #t "2023-02-23T13:18:22.976150Z[Etc/UTC]",
 :via
 [{:status :failed,
   :class clojure.lang.ExceptionInfo,
   :error "Unable to bin Field without a min/max value",
   :stacktrace
   ["--> query_processor.middleware.binning$update_binning_strategy_in_inner_query$replace_41646__41647.invoke(binning.clj:214)"
    "mbql.util.match.impl$replace_in_collection.invokeStatic(impl.cljc:47)"
    "mbql.util.match.impl$replace_in_collection.invoke(impl.cljc:38)"
    "query_processor.middleware.binning$update_binning_strategy_in_inner_query$replace_41646__41647.invoke(binning.clj:214)"
    "mbql.util.match.impl$replace_in_collection$iter__23578__23582$fn__23583.invoke(impl.cljc:44)"
    "mbql.util.match.impl$replace_in_collection.invokeStatic(impl.cljc:43)"
    "mbql.util.match.impl$replace_in_collection.invoke(impl.cljc:38)"
    "query_processor.middleware.binning$update_binning_strategy_in_inner_query$replace_41646__41647.invoke(binning.clj:214)"
    "query_processor.middleware.binning$update_binning_strategy_in_inner_query.invokeStatic(binning.clj:214)"
    "query_processor.middleware.binning$update_binning_strategy_in_inner_query.invoke(binning.clj:210)"
    "query_processor.middleware.binning$update_binning_strategy.invokeStatic(binning.clj:228)"
    "query_processor.middleware.binning$update_binning_strategy.invoke(binning.clj:221)"
    "query_processor$preprocess_STAR_$fn__55068.invoke(query_processor.clj:127)"
    "query_processor$preprocess_STAR_.invokeStatic(query_processor.clj:125)"
    "query_processor$preprocess_STAR_.invoke(query_processor.clj:120)"
    "query_processor$fn__55076$combined_pre_process__55077$combined_pre_process_STAR___55078.invoke(query_processor.clj:209)"
    "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__53521$fn__53526.invoke(resolve_database_and_driver.clj:35)"
    "driver$do_with_driver.invokeStatic(driver.clj:76)"
    "driver$do_with_driver.invoke(driver.clj:72)"
    "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__53521.invoke(resolve_database_and_driver.clj:34)"
    "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__49525.invoke(fetch_source_query.clj:314)"
    "query_processor.middleware.store$initialize_store$fn__49715$fn__49716.invoke(store.clj:11)"
    "query_processor.store$do_with_store.invokeStatic(store.clj:45)"
    "query_processor.store$do_with_store.invoke(store.clj:39)"
    "query_processor.middleware.store$initialize_store$fn__49715.invoke(store.clj:10)"
    "query_processor.middleware.normalize_query$normalize$fn__53793.invoke(normalize_query.clj:22)"
    "query_processor.middleware.constraints$add_default_userland_constraints$fn__50803.invoke(constraints.clj:53)"
    "query_processor.middleware.process_userland_query$process_userland_query$fn__53732.invoke(process_userland_query.clj:145)"
    "query_processor.middleware.catch_exceptions$catch_exceptions$fn__54104.invoke(catch_exceptions.clj:167)"
    "query_processor.reducible$async_qp$qp_STAR___45514$thunk__45516.invoke(reducible.clj:100)"
    "query_processor.reducible$async_qp$qp_STAR___45514.invoke(reducible.clj:106)"
    "query_processor.reducible$async_qp$qp_STAR___45514.invoke(reducible.clj:91)"
    "query_processor.reducible$sync_qp$qp_STAR___45525.doInvoke(reducible.clj:126)"
    "query_processor$process_userland_query.invokeStatic(query_processor.clj:331)"
    "query_processor$process_userland_query.doInvoke(query_processor.clj:327)"
    "query_processor$fn__55123$process_query_and_save_execution_BANG___55132$fn__55135.invoke(query_processor.clj:342)"
    "query_processor$fn__55123$process_query_and_save_execution_BANG___55132.invoke(query_processor.clj:335)"
    "query_processor$fn__55167$process_query_and_save_with_max_results_constraints_BANG___55176$fn__55179.invoke(query_processor.clj:354)"
    "query_processor$fn__55167$process_query_and_save_with_max_results_constraints_BANG___55176.invoke(query_processor.clj:347)"
    "api.dataset$run_query_async$fn__70529.invoke(dataset.clj:69)"
    "query_processor.streaming$streaming_response_STAR_$fn__40610$fn__40611.invoke(streaming.clj:162)"
    "query_processor.streaming$streaming_response_STAR_$fn__40610.invoke(streaming.clj:161)"
    "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:65)"
    "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:63)"
    "async.streaming_response$do_f_async$task__20021.invoke(streaming_response.clj:84)"],
   :ex-data {:clause [:field 2405 {:binning {:strategy :default}}]}}],
 :error_type :invalid-query,
 :json_query
 {:type "query",
  :query
  {:source-table 146,
   :filter ["time-interval" ["field" 2414 nil] "current" "week"],
   :aggregation [["count"]],
   :breakout [["field" 2405 {:binning {:strategy "default"}}]]},
  :database 3,
  :parameters [],
  :middleware {:js-int-to-string? true, :add-default-userland-constraints? true}},
 :native nil,
 :status :failed,
 :class clojure.lang.ExceptionInfo,
 :stacktrace
 ["--> query_processor.middleware.binning$fn__41361$extract_bounds__41366$fn__41367.invoke(binning.clj:49)"
  "query_processor.middleware.binning$fn__41361$extract_bounds__41366.invoke(binning.clj:30)"
  "query_processor.middleware.binning$fn__41604$update_binned_field__41609$fn__41622.invoke(binning.clj:197)"
  "query_processor.middleware.binning$fn__41604$update_binned_field__41609.invoke(binning.clj:189)"
  "query_processor.middleware.binning$update_binning_strategy_in_inner_query$replace_41646__41647.invoke(binning.clj:214)"
  "mbql.util.match.impl$replace_in_collection.invokeStatic(impl.cljc:47)"
  "mbql.util.match.impl$replace_in_collection.invoke(impl.cljc:38)"
  "query_processor.middleware.binning$update_binning_strategy_in_inner_query$replace_41646__41647.invoke(binning.clj:214)"
  "mbql.util.match.impl$replace_in_collection$iter__23578__23582$fn__23583.invoke(impl.cljc:44)"
  "mbql.util.match.impl$replace_in_collection.invokeStatic(impl.cljc:43)"
  "mbql.util.match.impl$replace_in_collection.invoke(impl.cljc:38)"
  "query_processor.middleware.binning$update_binning_strategy_in_inner_query$replace_41646__41647.invoke(binning.clj:214)"
  "query_processor.middleware.binning$update_binning_strategy_in_inner_query.invokeStatic(binning.clj:214)"
  "query_processor.middleware.binning$update_binning_strategy_in_inner_query.invoke(binning.clj:210)"
  "query_processor.middleware.binning$update_binning_strategy.invokeStatic(binning.clj:228)"
  "query_processor.middleware.binning$update_binning_strategy.invoke(binning.clj:221)"
  "query_processor$preprocess_STAR_$fn__55068.invoke(query_processor.clj:127)"
  "query_processor$preprocess_STAR_.invokeStatic(query_processor.clj:125)"
  "query_processor$preprocess_STAR_.invoke(query_processor.clj:120)"
  "query_processor$fn__55076$combined_pre_process__55077$combined_pre_process_STAR___55078.invoke(query_processor.clj:209)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__53521$fn__53526.invoke(resolve_database_and_driver.clj:35)"
  "driver$do_with_driver.invokeStatic(driver.clj:76)"
  "driver$do_with_driver.invoke(driver.clj:72)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__53521.invoke(resolve_database_and_driver.clj:34)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__49525.invoke(fetch_source_query.clj:314)"
  "query_processor.middleware.store$initialize_store$fn__49715$fn__49716.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:45)"
  "query_processor.store$do_with_store.invoke(store.clj:39)"
  "query_processor.middleware.store$initialize_store$fn__49715.invoke(store.clj:10)"
  "query_processor.middleware.normalize_query$normalize$fn__53793.invoke(normalize_query.clj:22)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__50803.invoke(constraints.clj:53)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__53732.invoke(process_userland_query.clj:145)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__54104.invoke(catch_exceptions.clj:167)"
  "query_processor.reducible$async_qp$qp_STAR___45514$thunk__45516.invoke(reducible.clj:100)"
  "query_processor.reducible$async_qp$qp_STAR___45514.invoke(reducible.clj:106)"
  "query_processor.reducible$async_qp$qp_STAR___45514.invoke(reducible.clj:91)"
  "query_processor.reducible$sync_qp$qp_STAR___45525.doInvoke(reducible.clj:126)"
  "query_processor$process_userland_query.invokeStatic(query_processor.clj:331)"
  "query_processor$process_userland_query.doInvoke(query_processor.clj:327)"
  "query_processor$fn__55123$process_query_and_save_execution_BANG___55132$fn__55135.invoke(query_processor.clj:342)"
  "query_processor$fn__55123$process_query_and_save_execution_BANG___55132.invoke(query_processor.clj:335)"
  "query_processor$fn__55167$process_query_and_save_with_max_results_constraints_BANG___55176$fn__55179.invoke(query_processor.clj:354)"
  "query_processor$fn__55167$process_query_and_save_with_max_results_constraints_BANG___55176.invoke(query_processor.clj:347)"
  "api.dataset$run_query_async$fn__70529.invoke(dataset.clj:69)"
  "query_processor.streaming$streaming_response_STAR_$fn__40610$fn__40611.invoke(streaming.clj:162)"
  "query_processor.streaming$streaming_response_STAR_$fn__40610.invoke(streaming.clj:161)"
  "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:65)"
  "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:63)"
  "async.streaming_response$do_f_async$task__20021.invoke(streaming_response.clj:84)"],
 :card_id nil,
 :context :ad-hoc,
 :error "Unable to bin Field without a min/max value",
 :row_count 0,
 :running_time 0,
 :preprocessed nil,
 :ex-data {:type :invalid-query, :field-id 2405, :fingerprint nil},
 :data {:rows [], :cols []}}

[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:18:23+01:00 DEBUG metabase.server.middleware.log POST /api/dataset 202 [ASYNC: completed] 159.9 ms (16 DB calls) App DB connections: 0/15 Jetty threads: 2/50 (5 idle, 0 queued) (98 total active threads) Queries in flight: 1 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:23+01:00 INFO metabase.server.middleware.exceptions Request canceled before finishing.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 INFO metabase.server.middleware.exceptions Request canceled before finishing.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 DEBUG metabase.server.middleware.log GET /api/session/properties 200 14.7 ms (4 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (3 idle, 0 queued) (94 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 DEBUG metabase.server.middleware.log GET /api/user/current 200 60.9 ms (9 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (3 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 DEBUG metabase.server.middleware.log GET /api/collection/root 200 4.7 ms (2 DB calls) App DB connections: 3/15 Jetty threads: 6/50 (1 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 DEBUG metabase.server.middleware.log GET /api/bookmark 200 3.5 ms (1 DB calls) App DB connections: 3/15 Jetty threads: 6/50 (1 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 DEBUG metabase.server.middleware.log GET /api/database 200 8.8 ms (3 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (3 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 DEBUG metabase.server.middleware.log GET /api/timeline 200 5.2 ms (3 DB calls) App DB connections: 3/15 Jetty threads: 6/50 (1 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 DEBUG metabase.server.middleware.log GET /api/database/3/schemas 200 8.8 ms (4 DB calls) App DB connections: 3/15 Jetty threads: 5/50 (2 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 DEBUG metabase.server.middleware.log GET /api/table/197/query_metadata 200 18.2 ms (9 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (3 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 DEBUG metabase.server.middleware.log GET /api/collection/tree 200 78.5 ms (43 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 DEBUG metabase.server.middleware.log POST /api/dataset 202 [ASYNC: completed] 420.4 ms (22 DB calls) App DB connections: 1/15 Jetty threads: 2/50 (5 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:24+01:00 INFO metabase.server.middleware.exceptions Request canceled before finishing.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:27+01:00 INFO metabase.server.middleware.exceptions Request canceled before finishing.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:29+01:00 DEBUG metabase.server.middleware.log GET /api/session/properties 200 13.0 ms (4 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (3 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:29+01:00 DEBUG metabase.server.middleware.log GET /api/user/current 200 18.8 ms (9 DB calls) App DB connections: 0/15 Jetty threads: 4/50 (3 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:29+01:00 DEBUG metabase.server.middleware.log GET /api/collection/root 200 4.1 ms (2 DB calls) App DB connections: 2/15 Jetty threads: 7/50 (1 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:29+01:00 DEBUG metabase.server.middleware.log GET /api/bookmark 200 11.6 ms (1 DB calls) App DB connections: 2/15 Jetty threads: 5/50 (2 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:29+01:00 DEBUG metabase.server.middleware.log GET /api/database 200 14.7 ms (3 DB calls) App DB connections: 0/15 Jetty threads: 4/50 (3 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:29+01:00 DEBUG metabase.server.middleware.log GET /api/timeline 200 5.0 ms (3 DB calls) App DB connections: 2/15 Jetty threads: 5/50 (2 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:29+01:00 DEBUG metabase.server.middleware.log GET /api/database/3/schemas 200 9.5 ms (4 DB calls) App DB connections: 2/15 Jetty threads: 5/50 (2 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:29+01:00 DEBUG metabase.server.middleware.log GET /api/table/198/query_metadata 200 18.1 ms (9 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (3 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:29+01:00 DEBUG metabase.server.middleware.log GET /api/collection/tree 200 94.0 ms (43 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:29+01:00 INFO metabase.server.middleware.exceptions Request canceled before finishing.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 WARN metabase.driver.bigquery-cloud-sdk.query-processor Warning: missing type mapping for parsing BigQuery results of type STRING.
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:19:30+01:00 DEBUG metabase.server.middleware.log POST /api/dataset 202 [ASYNC: completed] 513.0 ms (23 DB calls) App DB connections: 1/15 Jetty threads: 2/50 (5 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:21:17+01:00 DEBUG metabase.server.middleware.log GET /api/setting 200 29.6 ms (3 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (3 idle, 0 queued) (94 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:21:17+01:00 DEBUG metabase.server.middleware.log GET /api/session/properties 200 38.9 ms (4 DB calls) App DB connections: 0/15 Jetty threads: 4/50 (3 idle, 0 queued) (94 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:21:17+01:00 DEBUG metabase.server.middleware.log GET /api/setup/admin_checklist 200 24.2 ms (11 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:21:18+01:00 DEBUG metabase.server.middleware.log GET /api/util/bug_report_details 200 5.1 ms (1 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:21:55+01:00 DEBUG metabase.server.middleware.log GET /api/collection 200 10.3 ms (4 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:21:55+01:00 DEBUG metabase.server.middleware.log GET /api/collection/graph 200 49.8 ms (5 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:23:13+01:00 DEBUG metabase.server.middleware.log GET /api/database 200 6.3 ms (3 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (94 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:23:14+01:00 DEBUG metabase.server.middleware.log GET /api/database/3/metadata 200 370.0 ms (146 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (94 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:23:14+01:00 DEBUG metabase.server.middleware.log GET /api/setting 200 97.0 ms (3 DB calls) App DB connections: 1/15 Jetty threads: 6/50 (1 idle, 0 queued) (94 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:23:14+01:00 DEBUG metabase.server.middleware.log GET /api/session/properties 200 104.5 ms (4 DB calls) App DB connections: 1/15 Jetty threads: 6/50 (1 idle, 0 queued) (95 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:23:14+01:00 DEBUG metabase.server.middleware.log GET /api/setup/admin_checklist 200 204.6 ms (11 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (3 idle, 0 queued) (98 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:23:20+01:00 DEBUG metabase.server.middleware.log GET /api/table/146 200 9.3 ms (5 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (98 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:23:20+01:00 DEBUG metabase.server.middleware.log GET /api/database/3/idfields 200 27.0 ms (6 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (3 idle, 0 queued) (98 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:23:20+01:00 DEBUG metabase.server.middleware.log GET /api/table/146/query_metadata 200 28.1 ms (9 DB calls) App DB connections: 0/15 Jetty threads: 4/50 (3 idle, 0 queued) (98 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:23:27+01:00 DEBUG metabase.server.middleware.log GET /api/field/2424/values 200 12.4 ms (4 DB calls) App DB connections: 1/15 Jetty threads: 5/50 (2 idle, 0 queued) (98 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:23:27+01:00 DEBUG metabase.server.middleware.log GET /api/table/146/query_metadata 200 21.5 ms (9 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (3 idle, 0 queued) (98 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:23:28+01:00 DEBUG metabase.server.middleware.log GET /api/database/3/metadata 200 381.1 ms (146 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (98 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:01+01:00 DEBUG metabase.server.middleware.log GET /api/util/bug_report_details 200 2.8 ms (1 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (94 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:24+01:00 DEBUG metabase.server.middleware.log GET /api/session/properties 200 16.7 ms (4 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (3 idle, 0 queued) (94 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:24+01:00 DEBUG metabase.server.middleware.log GET /api/user/current 200 49.2 ms (9 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (94 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:24+01:00 DEBUG metabase.server.middleware.log GET /api/collection/root 200 4.1 ms (2 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (4 idle, 0 queued) (94 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:24+01:00 DEBUG metabase.server.middleware.log GET /api/bookmark 200 7.1 ms (1 DB calls) App DB connections: 4/15 Jetty threads: 7/50 (2 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:24+01:00 DEBUG metabase.server.middleware.log GET /api/database 200 14.1 ms (3 DB calls) App DB connections: 4/15 Jetty threads: 6/50 (3 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:24+01:00 DEBUG metabase.server.middleware.log GET /api/collection/tree 200 71.7 ms (43 DB calls) App DB connections: 2/15 Jetty threads: 5/50 (4 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:24+01:00 DEBUG metabase.server.middleware.log GET /api/activity/recent_views 200 118.6 ms (8 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (5 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:24+01:00 DEBUG metabase.server.middleware.log GET /api/activity/popular_items 200 157.4 ms (10 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (6 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:28+01:00 DEBUG metabase.server.middleware.log GET /api/collection/37 200 8.1 ms (5 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (5 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:28+01:00 DEBUG metabase.server.middleware.log GET /api/collection/tree 200 72.3 ms (43 DB calls) App DB connections: 2/15 Jetty threads: 5/50 (4 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:28+01:00 DEBUG metabase.server.middleware.log GET /api/collection/37/items 200 24.1 ms (5 DB calls) App DB connections: 0/15 Jetty threads: 4/50 (5 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:28+01:00 DEBUG metabase.server.middleware.log GET /api/collection/37/items 200 70.0 ms (5 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (6 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:28+01:00 DEBUG metabase.server.middleware.log GET /api/collection/42 200 9.3 ms (5 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (6 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:28+01:00 DEBUG metabase.server.middleware.log GET /api/collection/42/items 200 23.2 ms (5 DB calls) App DB connections: 0/15 Jetty threads: 4/50 (5 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:28+01:00 DEBUG metabase.server.middleware.log GET /api/collection/42/items 200 43.1 ms (5 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (6 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:31+01:00 DEBUG metabase.server.middleware.log GET /api/timeline 200 5.8 ms (3 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (5 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:31+01:00 DEBUG metabase.server.middleware.log GET /api/card/802 200 19.0 ms (10 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (6 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:31+01:00 DEBUG metabase.server.middleware.log GET /api/alert/question/802 200 2.4 ms (1 DB calls) App DB connections: 2/15 Jetty threads: 5/50 (4 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:31+01:00 DEBUG metabase.server.middleware.log GET /api/database/3/schemas 200 9.3 ms (4 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (5 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:31+01:00 DEBUG metabase.server.middleware.log GET /api/table/146/query_metadata 200 18.2 ms (9 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (6 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:31+01:00 ERROR metabase.query-processor.middleware.catch-exceptions Error processing query: Unable to bin Field without a min/max value
{:database_id 3,
 :started_at #t "2023-02-23T13:25:31.216300Z[Etc/UTC]",
 :via
 [{:status :failed,
   :class clojure.lang.ExceptionInfo,
   :error "Unable to bin Field without a min/max value",
   :stacktrace
   ["--> query_processor.middleware.binning$update_binning_strategy_in_inner_query$replace_41646__41647.invoke(binning.clj:214)"
    "mbql.util.match.impl$replace_in_collection.invokeStatic(impl.cljc:47)"
    "mbql.util.match.impl$replace_in_collection.invoke(impl.cljc:38)"
    "query_processor.middleware.binning$update_binning_strategy_in_inner_query$replace_41646__41647.invoke(binning.clj:214)"
    "mbql.util.match.impl$replace_in_collection$iter__23578__23582$fn__23583.invoke(impl.cljc:44)"
    "mbql.util.match.impl$replace_in_collection.invokeStatic(impl.cljc:43)"
    "mbql.util.match.impl$replace_in_collection.invoke(impl.cljc:38)"
    "query_processor.middleware.binning$update_binning_strategy_in_inner_query$replace_41646__41647.invoke(binning.clj:214)"
    "query_processor.middleware.binning$update_binning_strategy_in_inner_query.invokeStatic(binning.clj:214)"
    "query_processor.middleware.binning$update_binning_strategy_in_inner_query.invoke(binning.clj:210)"
    "query_processor.middleware.binning$update_binning_strategy.invokeStatic(binning.clj:228)"
    "query_processor.middleware.binning$update_binning_strategy.invoke(binning.clj:221)"
    "query_processor$preprocess_STAR_$fn__55068.invoke(query_processor.clj:127)"
    "query_processor$preprocess_STAR_.invokeStatic(query_processor.clj:125)"
    "query_processor$preprocess_STAR_.invoke(query_processor.clj:120)"
    "query_processor$fn__55076$combined_pre_process__55077$combined_pre_process_STAR___55078.invoke(query_processor.clj:209)"
    "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__53521$fn__53526.invoke(resolve_database_and_driver.clj:35)"
    "driver$do_with_driver.invokeStatic(driver.clj:76)"
    "driver$do_with_driver.invoke(driver.clj:72)"
    "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__53521.invoke(resolve_database_and_driver.clj:34)"
    "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__49525.invoke(fetch_source_query.clj:314)"
    "query_processor.middleware.store$initialize_store$fn__49715$fn__49716.invoke(store.clj:11)"
    "query_processor.store$do_with_store.invokeStatic(store.clj:45)"
    "query_processor.store$do_with_store.invoke(store.clj:39)"
    "query_processor.middleware.store$initialize_store$fn__49715.invoke(store.clj:10)"
    "query_processor.middleware.normalize_query$normalize$fn__53793.invoke(normalize_query.clj:22)"
    "query_processor.middleware.constraints$add_default_userland_constraints$fn__50803.invoke(constraints.clj:53)"
    "query_processor.middleware.process_userland_query$process_userland_query$fn__53732.invoke(process_userland_query.clj:145)"
    "query_processor.middleware.catch_exceptions$catch_exceptions$fn__54104.invoke(catch_exceptions.clj:167)"
    "query_processor.reducible$async_qp$qp_STAR___45514$thunk__45516.invoke(reducible.clj:100)"
    "query_processor.reducible$async_qp$qp_STAR___45514$fn__45518.invoke(reducible.clj:105)"],
   :ex-data {:clause [:field 2424 {:binning {:strategy :default}}]}}],
 :error_type :invalid-query,
 :json_query
 {:constraints {:max-results 10000, :max-results-bare-rows 2000},
  :type :query,
  :middleware {:js-int-to-string? true, :ignore-cached-results? false, :process-viz-settings? false},
  :database 3,
  :query
  {:source-table 146,
   :filter [:time-interval [:field 2414 nil] :current :week],
   :aggregation [[:count]],
   :breakout [[:field 2424 {:binning {:strategy :default}}]]},
  :parameters [],
  :async? true,
  :cache-ttl nil},
 :native nil,
 :status :failed,
 :class clojure.lang.ExceptionInfo,
 :stacktrace
 ["--> query_processor.middleware.binning$fn__41361$extract_bounds__41366$fn__41367.invoke(binning.clj:49)"
  "query_processor.middleware.binning$fn__41361$extract_bounds__41366.invoke(binning.clj:30)"
  "query_processor.middleware.binning$fn__41604$update_binned_field__41609$fn__41622.invoke(binning.clj:197)"
  "query_processor.middleware.binning$fn__41604$update_binned_field__41609.invoke(binning.clj:189)"
  "query_processor.middleware.binning$update_binning_strategy_in_inner_query$replace_41646__41647.invoke(binning.clj:214)"
  "mbql.util.match.impl$replace_in_collection.invokeStatic(impl.cljc:47)"
  "mbql.util.match.impl$replace_in_collection.invoke(impl.cljc:38)"
  "query_processor.middleware.binning$update_binning_strategy_in_inner_query$replace_41646__41647.invoke(binning.clj:214)"
  "mbql.util.match.impl$replace_in_collection$iter__23578__23582$fn__23583.invoke(impl.cljc:44)"
  "mbql.util.match.impl$replace_in_collection.invokeStatic(impl.cljc:43)"
  "mbql.util.match.impl$replace_in_collection.invoke(impl.cljc:38)"
  "query_processor.middleware.binning$update_binning_strategy_in_inner_query$replace_41646__41647.invoke(binning.clj:214)"
  "query_processor.middleware.binning$update_binning_strategy_in_inner_query.invokeStatic(binning.clj:214)"
  "query_processor.middleware.binning$update_binning_strategy_in_inner_query.invoke(binning.clj:210)"
  "query_processor.middleware.binning$update_binning_strategy.invokeStatic(binning.clj:228)"
  "query_processor.middleware.binning$update_binning_strategy.invoke(binning.clj:221)"
  "query_processor$preprocess_STAR_$fn__55068.invoke(query_processor.clj:127)"
  "query_processor$preprocess_STAR_.invokeStatic(query_processor.clj:125)"
  "query_processor$preprocess_STAR_.invoke(query_processor.clj:120)"
  "query_processor$fn__55076$combined_pre_process__55077$combined_pre_process_STAR___55078.invoke(query_processor.clj:209)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__53521$fn__53526.invoke(resolve_database_and_driver.clj:35)"
  "driver$do_with_driver.invokeStatic(driver.clj:76)"
  "driver$do_with_driver.invoke(driver.clj:72)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__53521.invoke(resolve_database_and_driver.clj:34)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__49525.invoke(fetch_source_query.clj:314)"
  "query_processor.middleware.store$initialize_store$fn__49715$fn__49716.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:45)"
  "query_processor.store$do_with_store.invoke(store.clj:39)"
  "query_processor.middleware.store$initialize_store$fn__49715.invoke(store.clj:10)"
  "query_processor.middleware.normalize_query$normalize$fn__53793.invoke(normalize_query.clj:22)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__50803.invoke(constraints.clj:53)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__53732.invoke(process_userland_query.clj:145)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__54104.invoke(catch_exceptions.clj:167)"
  "query_processor.reducible$async_qp$qp_STAR___45514$thunk__45516.invoke(reducible.clj:100)"
  "query_processor.reducible$async_qp$qp_STAR___45514$fn__45518.invoke(reducible.clj:105)"],
 :card_id 802,
 :context :question,
 :error "Unable to bin Field without a min/max value",
 :row_count 0,
 :running_time 0,
 :preprocessed nil,
 :ex-data {:type :invalid-query, :field-id 2424, :fingerprint nil},
 :data {:rows [], :cols []}}

[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:31+01:00 DEBUG metabase.server.middleware.log POST /api/card/802/query 202 [ASYNC: completed] 122.9 ms (17 DB calls) App DB connections: 0/15 Jetty threads: 2/50 (7 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:34+01:00 DEBUG metabase.server.middleware.log GET /api/session/properties 200 21.2 ms (4 DB calls) App DB connections: 1/15 Jetty threads: 4/50 (5 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:34+01:00 DEBUG metabase.server.middleware.log GET /api/user/current 200 26.4 ms (9 DB calls) App DB connections: 0/15 Jetty threads: 4/50 (5 idle, 0 queued) (96 total active threads) Queries in flight: 0 (0 queued)
[a5635773-5146-4999-9851-babfc05f5228] 2023-02-23T14:25:34+01:00 DEBUG metabase.server.middleware.log GET /api/database 200 5.8 ms (3 DB calls) App DB connections: 0/15 Jetty threads: 3/50 (6 idle, 0 queued) (97 total active threads) Queries in flight: 0 (0 queued)

```


**To Reproduce**
Steps to reproduce the behavior (if you can reproduce the bug using the Sample Database, we will find the issue faster):
1. Go to any of my tables.
2. Look for a numeric column
3. On the column header click on the dropdown and select Distribution
4. See error

**Expected behavior**
Histrogram of auto-binned value shows.

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-GB",
    "platform": "MacIntel",
    "userAgent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/110.0.0.0 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "ANSI_X3.4-1968",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.18+10-post-Debian-1deb10u1",
    "java.vendor": "Debian",
    "java.vendor.url": "https://tracker.debian.org/openjdk-11",
    "java.version": "11.0.18",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.18+10-post-Debian-1deb10u1",
    "os.name": "Linux",
    "os.version": "5.4.202+",
    "user.language": "en",
    "user.timezone": "Etc/UTC"
  },
  "metabase-info": {
    "databases": [
      "bigquery-cloud-sdk"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "13.7"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.5.0"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2023-02-19",
      "tag": "v0.45.3",
      "branch": "release-x.45.x",
      "hash": "070f57b"
    },
    "settings": {
      "report-timezone": null
    }
  }
}
```

**Severity**
Significantly increases time to create histograms to an extend where you would rather not spend that time.

**Additional context**
Could be linked to the fact that the type of all columns are bignumerics as this led to issues before: https://github.com/metabase/metabase/issues/25505

--- a/modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj
+++ b/modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj
@@ -998,3 +998,29 @@
                   qp.compile/compile-with-inline-parameters
                   :query
                   pretty-sql-lines))))))
+
+(deftest fingerprint-and-bin-bigdecimal-test
+  (mt/test-driver
+    :bigquery-cloud-sdk
+    (mt/dataset
+      (mt/dataset-definition
+        "bigthings"
+        ["bigthings"
+         [{:field-name "bd" :base-type :type/Decimal}]
+         [[12345678901234567890.1234567890M]
+          [22345678901234567890.1234567890M]
+          [32345678901234567890.1234567890M]]])
+
+      ;; Must sync field values
+      (sync/sync-database! (mt/db))
+      (is (= "BIGNUMERIC"
+             (t2/select-one-fn :database_type :model/Field :id (mt/id :bigthings :bd))))
+      (is (= [[12000000000000000000M 1]
+              [21000000000000000000M 1]
+              [30000000000000000000M 1]]
+             (mt/rows
+               (mt/run-mbql-query bigthings
+                                  {:aggregation [[:count]]
+                                   :breakout [[:field %bigthings.bd
+                                               {:type :type/Decimal
+                                                :binning {:strategy "default"}}]]})))))))

--- a/modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj
+++ b/modules/drivers/bigquery-cloud-sdk/test/metabase/driver/bigquery_cloud_sdk_test.clj
@@ -1004,12 +1004,12 @@
     :bigquery-cloud-sdk
     (mt/dataset
       (mt/dataset-definition
-        "bigthings"
-        ["bigthings"
-         [{:field-name "bd" :base-type :type/Decimal}]
-         [[12345678901234567890.1234567890M]
-          [22345678901234567890.1234567890M]
-          [32345678901234567890.1234567890M]]])
+       "bigthings"
+       ["bigthings"
+        [{:field-name "bd" :base-type :type/Decimal}]
+        [[12345678901234567890.1234567890M]
+         [22345678901234567890.1234567890M]
+         [32345678901234567890.1234567890M]]])
 
       ;; Must sync field values
       (sync/sync-database! (mt/db))
@@ -1019,8 +1019,8 @@
               [21000000000000000000M 1]
               [30000000000000000000M 1]]
              (mt/rows
-               (mt/run-mbql-query bigthings
-                                  {:aggregation [[:count]]
-                                   :breakout [[:field %bigthings.bd
-                                               {:type :type/Decimal
-                                                :binning {:strategy "default"}}]]})))))))
+              (mt/run-mbql-query bigthings
+                {:aggregation [[:count]]
+                 :breakout [[:field %bigthings.bd
+                             {:type :type/Decimal
+                              :binning {:strategy "default"}}]]})))))))

--- a/modules/drivers/bigquery-cloud-sdk/test/metabase/test/data/bigquery_cloud_sdk.clj
+++ b/modules/drivers/bigquery-cloud-sdk/test/metabase/test/data/bigquery_cloud_sdk.clj
@@ -125,7 +125,7 @@
                :type/Date           :DATE
                :type/DateTime       :DATETIME
                :type/DateTimeWithTZ :TIMESTAMP
-               :type/Decimal        :NUMERIC
+               :type/Decimal        :BIGNUMERIC
                :type/Dictionary     :RECORD
                :type/Float          :FLOAT
                :type/Integer        :INTEGER

